webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+package io.kestra.webserver.services.ai.gemini;
+
+import com.devskiller.friendly_id.FriendlyId;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.adk.agents.*;
+import com.google.adk.events.Event;
+import com.google.adk.models.Gemini;
+import com.google.adk.models.LlmResponse;
+import com.google.adk.runner.InMemoryRunner;
+import com.google.adk.sessions.Session;
+import com.google.genai.Client;
+import com.google.genai.types.*;
+import io.kestra.core.docs.JsonSchemaGenerator;
+import io.kestra.core.exceptions.AiException;
+import io.kestra.core.exceptions.KestraException;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.validations.ValidateConstraintViolation;
+import io.kestra.core.plugins.PluginRegistry;
+import io.kestra.core.secret.SecretPluginInterface;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.core.services.FlowService;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.tenant.TenantService;
+import io.kestra.webserver.models.ai.FlowGenerationPrompt;
+import io.kestra.webserver.services.ai.AiServiceInterface;
+import io.micronaut.context.annotation.Requires;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Singleton
+@Requires(property = "kestra.ai.type", value = "gemini")
+@Requires(property = "kestra.ai.gemini.api-key")
+@Slf4j
+public class GeminiAiService implements AiServiceInterface {
+    private static final String KESTRA_FLOW_AGENT_APP_ID = "kestra_flow_agent";
+    private static final String KESTRA_FLOW_BUILDER_NAME = "kestra_flow_builder_agent";
+    private static final String ALREADY_VALID_FLOW = "Sorry, your flow seems to already do that, please try again with extra instructions if you wish me to do some modifications.";
```

Comment:
```suggestion
    private static final String ALREADY_VALID_FLOW = "This flow already performs the requested action. Please provide additional instructions if you would like to request modifications.";
```

You're too nice, Brian. No need to say Sorry 🤗